internal/httpproxy/metrics.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Twingate Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package httpproxy
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8sgateway/internal/metrics"
+)
+
+const (
+	connCategoryUnknown = "unknown"
+	connCategoryProxy   = "proxy"
+	connCategoryHealth  = "health"
+)
+
+type proxyConnMetrics struct {
+	activeConn   prometheus.Gauge
+	connTotal    *prometheus.CounterVec
+	connDuration *prometheus.HistogramVec
+
+	connectTotal    *prometheus.CounterVec
+	connectDuration *prometheus.HistogramVec
+}
+
+func registerProxyConnMetrics(registry *prometheus.Registry) *proxyConnMetrics {
```

Comment:
Previously, we declared the metrics as shared global variables (see [diff](https://github.com/Twingate/kubernetes-access-gateway/pull/71/commits/b26799334fc4fc4ec27ff41c45295b467bb4b606)), which resulted in data races in test cases (see [here](https://github.com/Twingate/kubernetes-access-gateway/actions/runs/16491431219/job/46626802915)). We'd have different test case registering and writing to the shared metrics. Thus, I moved the metrics here to ensure each test case has its own metrics instance.